### ISSUE TITLE
Canu release v1.5 failed with:   didn't find any mers?

### ISSUE THREAD

**Issue by zya226** [marbl/canu#1335]
Hi Team Canu,

Same titled issue was found in posts. But my error report below is different. 
```
Use of uninitialized value $error[0] in join or string at /share/pkg/perl/5.18.1/lib/5.18.1/Carp.pm line 301.
```
================================================================================
```
Please panic.  Canu failed, and it shouldn't have.

Stack trace:

 at /share/pkg/canu/1.5/Linux-amd64/bin/lib/canu/Meryl.pm line 668.
        canu::Meryl::merylProcess('GD02', 'cor') called at /share/pkg/canu/1.5/Linux-amd64/bin/canu line 536


Last few lines of the relevant log file (correction/0-mercounts/GD02.ms16.histogram.info):

merylStreamReader()-- ERROR: ./GD02.ms16.mcidx is not a merylStream index file!
merylStreamReader()-- ERROR: ./GD02.ms16.mcdat is not a merylStream data file!

Canu release v1.5 failed with:
  didn't find any mers?
```

Here is my command line:  ```canu -correct -p GD02 -d GD02_assembly corMemory=19g genomeSize=52m minReadLength=2000 minOverlapLength=500 corOutCoverage=300 corMinCoverage=2 -pacbio-raw GD02.filtered_subreads.fq.gz```

Could you please help me look at it? Any help would be appreciated. 

Thanks in advance,
Yong

**Comment by skoren** [comment]
It looks like a previous step in the pipeline must have failed, there should be more information in ```correction/0-mercounts/meryl*out```). However, before you spend time trying to diagnose this run, version 1.5 is extremely old so I would instead download the 1.8 release and try the run from scratch in a new folder.

**Comment by zya226** [comment]
Hi Sergey,


Thank you for your quick response. Follow your suggestion, I have changed to canu v1.8. However, I got new error. 


 
-- BEGIN CORRECTION
 
--
 
--
 
-- Mhap precompute jobs failed, retry.
 
--   job correction/1-overlapper/blocks/000001.dat FAILED.
 
--   job correction/1-overlapper/blocks/000002.dat FAILED.
 
--   job correction/1-overlapper/blocks/000003.dat FAILED.
 
--   job correction/1-overlapper/blocks/000004.dat FAILED.
 
--   job correction/1-overlapper/blocks/000005.dat FAILED.
 
--   job correction/1-overlapper/blocks/000006.dat FAILED.
 
--   job correction/1-overlapper/blocks/000007.dat FAILED.
 
--   job correction/1-overlapper/blocks/000008.dat FAILED.
 
--   job correction/1-overlapper/blocks/000009.dat FAILED.
 
--   job correction/1-overlapper/blocks/000010.dat FAILED.
 
--   job correction/1-overlapper/blocks/000011.dat FAILED.
 
--   job correction/1-overlapper/blocks/000012.dat FAILED.
 
--   job correction/1-overlapper/blocks/000013.dat FAILED.
 
--   job correction/1-overlapper/blocks/000014.dat FAILED.
 
--   job correction/1-overlapper/blocks/000015.dat FAILED.
 
--   job correction/1-overlapper/blocks/000016.dat FAILED.
 
--   job correction/1-overlapper/blocks/000017.dat FAILED.
 
--   job correction/1-overlapper/blocks/000018.dat FAILED.
 
--   job correction/1-overlapper/blocks/000019.dat FAILED.
 
--   job correction/1-overlapper/blocks/000020.dat FAILED.
 
--   job correction/1-overlapper/blocks/000021.dat FAILED.
 
--   job correction/1-overlapper/blocks/000022.dat FAILED.
 
--   job correction/1-overlapper/blocks/000023.dat FAILED.
 
--   job correction/1-overlapper/blocks/000024.dat FAILED.
 
--   job correction/1-overlapper/blocks/000025.dat FAILED.
 
--   job correction/1-overlapper/blocks/000026.dat FAILED.
 
--   job correction/1-overlapper/blocks/000027.dat FAILED.
 
--   job correction/1-overlapper/blocks/000028.dat FAILED.
 
--   job correction/1-overlapper/blocks/000029.dat FAILED.
 
--   job correction/1-overlapper/blocks/000030.dat FAILED.
 
--   job correction/1-overlapper/blocks/000031.dat FAILED.
 
--   job correction/1-overlapper/blocks/000032.dat FAILED.
 
--   job correction/1-overlapper/blocks/000033.dat FAILED.
 
--   job correction/1-overlapper/blocks/000034.dat FAILED.
 
--



Below is my command: canu -correct -p GD02 -d GD02_assembly merylThreads=30 genomeSize=52m minReadLength=2000 minOverlapLength=500 corOutCoverage=300 corMinCoverage=2 -pacbio-raw GD02.filtered_subreads.fq.gz
Do you have any idea what is wrong? Thanks,


 ------------------ 原始邮件 ------------------
发件人: "Sergey Koren"<notifications@github.com>;
发送时间: 2019年4月21日(星期天) 凌晨1:32
收件人: "marbl/canu"<canu@noreply.github.com>;
抄送: "Yong Zhang"<zya226@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [marbl/canu] Canu release v1.5 failed with:   didn't find anymers? (#1335)




It looks like a previous step in the pipeline must have failed, there should be more information in correction/0-mercounts/meryl*out). However, before you spend time trying to diagnose this run, version 1.5 is extremely old so I would instead download the 1.8 release and try the run from scratch in a new folder.

**Comment by zya226** [comment]
Hi Sergey,


I checked correction/1-overlapper/precompute.1.out.
  
-f = ../../0-mercounts/GD02.ms16.ignore.gz
 
-h = false
 
-k = 16
 
-p = ./000001.input.fasta
 
-q = .
 
-s = 
 


 
Reading in filter file ../../0-mercounts/GD02.ms16.ignore.gz.
 
Read in values for repeat 52538 and 0
 
Warning, k-mer filter file has zero elements.
 
Initializing
 
Trying size 52538 and it was successfull
 
Initialized
 
Time (s) to read filter file: 0.845603202
 
Read in k-mer filter for sizes: [16]
 
Processing FASTA files for binary compression...
 
Current # sequences loaded and processed from file: 5000...
 
Current # sequences loaded and processed from file: 10000...
 
Current # sequences loaded and processed from file: 15000...
 
Current # sequences loaded and processed from file: 20000...
 
/farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/GD02/GD02_assembly/correction/1-overlapper/precompute.sh: line 306: 48855 Killed                  /share/pkg/jdk/1.8.0_77/bin/java -d64 -XX:ParallelGCThreads=4 -server -Xms10138m -Xmx10138m -jar $bin/../share/java/classes/mhap-2.1.3.jar --repeat-weight 0.9 --repeat-idf-scale 10 -k 16 --store-full-id --num-hashes 256 --num-min-matches 3 --ordered-sketch-size 1000 --ordered-kmer-size 14 --threshold 0.8 --filter-threshold 0.0000001 --min-olap-length 500 --num-threads 4 -f ../../0-mercounts/GD02.ms16.ignore.gz -p ./$job.input.fasta -q .
 
Mhap failed.

 
TERM_MEMLIMIT: job killed after reaching LSF memory usage limit.
 
Exited with exit code 1.



It looks like running out of memory. Is that right? How do I change it?




------------------ 原始邮件 ------------------
发件人: "Sergey Koren"<notifications@github.com>;
发送时间: 2019年4月21日(星期天) 凌晨1:32
收件人: "marbl/canu"<canu@noreply.github.com>;
抄送: "Yong Zhang"<zya226@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [marbl/canu] Canu release v1.5 failed with:   didn't find anymers? (#1335)




It looks like a previous step in the pipeline must have failed, there should be more information in correction/0-mercounts/meryl*out). However, before you spend time trying to diagnose this run, version 1.5 is extremely old so I would instead download the 1.8 release and try the run from scratch in a new folder.

**Comment by skoren** [comment]
It looks like it is currently trying to use 10gb, do you know what memory it was at when your grid killed it? This can happen on some systems due to JVM configurations, see issues #298 for a long discussion on possible options in the JVM that could cause this.

You can try just increasing the memory for the job by adding something like ```gridOptionscormhap=-M 20G``` (the exact memory will vary based on your grid's memory units, you can see how canu was requesting 10gb in the ```precompute.jobSubmit-*.sh file```). Remove the ```precompute.jobSubmit-*.sh``` file and re-launch with the added option.

**Comment by zya226** [comment]
Hi Sergey,


I changed the parameters in precompute.jobSubmit-*.sh file and ran smoothly. However, I have a new problem. It is said that failed to load corrected reads into store
  
Read correction jobs failed, tried 2 times, giving up.
 
--   job 2-correction/results/0001.cns FAILED.
 
--   job 2-correction/results/0002.cns FAILED.
 
--   job 2-correction/results/0003.cns FAILED.
 
--   job 2-correction/results/0004.cns FAILED.
 
--   job 2-correction/results/0005.cns FAILED.
 
--   job 2-correction/results/0006.cns FAILED.
 
--   job 2-correction/results/0007.cns FAILED.
 
--   job 2-correction/results/0008.cns FAILED.
 
--   job 2-correction/results/0009.cns FAILED.
 
--   job 2-correction/results/0010.cns FAILED.
 
--   job 2-correction/results/0011.cns FAILED.
 
--   job 2-correction/results/0012.cns FAILED.
 
--   job 2-correction/results/0013.cns FAILED.
 
--   job 2-correction/results/0014.cns FAILED.
 
--   job 2-correction/results/0015.cns FAILED.
 
--   job 2-correction/results/0016.cns FAILED.
 
--   job 2-correction/results/0017.cns FAILED.
 
--   job 2-correction/results/0018.cns FAILED.
 
--   job 2-correction/results/0019.cns FAILED.
 
--   job 2-correction/results/0020.cns FAILED.
 
--   job 2-correction/results/0021.cns FAILED.
 
--   job 2-correction/results/0022.cns FAILED.
 
--   job 2-correction/results/0023.cns FAILED.
 
--   job 2-correction/results/0024.cns FAILED.
 
--   job 2-correction/results/0025.cns FAILED.
 
--   job 2-correction/results/0026.cns FAILED.
 
--   job 2-correction/results/0027.cns FAILED.
 
--   job 2-correction/results/0028.cns FAILED.
 
--   job 2-correction/results/0029.cns FAILED.
 
--   job 2-correction/results/0030.cns FAILED.
 
--   job 2-correction/results/0031.cns FAILED.
 
--   job 2-correction/results/0032.cns FAILED.
 
--   job 2-correction/results/0033.cns FAILED.
 
--   job 2-correction/results/0034.cns FAILED.
 
--   job 2-correction/results/0035.cns FAILED.
 
--   job 2-correction/results/0036.cns FAILED.



Do you have any idea?




------------------ 原始邮件 ------------------
发件人: "Sergey Koren"<notifications@github.com>;
发送时间: 2019年4月21日(星期天) 上午9:45
收件人: "marbl/canu"<canu@noreply.github.com>;
抄送: "Yong Zhang"<zya226@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [marbl/canu] Canu release v1.5 failed with:   didn't find anymers? (#1335)




It looks like it is currently trying to use 10gb, do you know what memory it was at when your grid killed it? This can happen on some systems due to JVM configurations, see issues #298 for a long discussion on possible options in the JVM that could cause this.
 
You can try just increasing the memory for the job by adding something like gridOptionscormhap=-M 20G (the exact memory will vary based on your grid's memory units, you can see how canu was requesting 10gb in the precompute.jobSubmit-*.sh file). Remove the precompute.jobSubmit-*.sh file and re-launch with the added option.

**Comment by zya226** [comment]
Sergey,


I have checked results folder in 2-correction, multiple *.cns and *.err are there. When I type tail -n 1 *err, it is not bye and give me 932338    4456        6    304-4420  (  4150) memory act          0 est  564582074 act/est 0.00
 







------------------ 原始邮件 ------------------
发件人: "Sergey Koren"<notifications@github.com>;
发送时间: 2019年4月21日(星期天) 上午9:45
收件人: "marbl/canu"<canu@noreply.github.com>;
抄送: "Yong Zhang"<zya226@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [marbl/canu] Canu release v1.5 failed with:   didn't find anymers? (#1335)




It looks like it is currently trying to use 10gb, do you know what memory it was at when your grid killed it? This can happen on some systems due to JVM configurations, see issues #298 for a long discussion on possible options in the JVM that could cause this.
 
You can try just increasing the memory for the job by adding something like gridOptionscormhap=-M 20G (the exact memory will vary based on your grid's memory units, you can see how canu was requesting 10gb in the precompute.jobSubmit-*.sh file). Remove the precompute.jobSubmit-*.sh file and re-launch with the added option.

**Comment by skoren** [comment]
Those errors indicate the generation of the corrected reads failed, not the loading itself. Any cns file with WORKING in the file name are incomplete. Did these jobs get killed for exceeding memory? That's the likeliest issue, try doubling whatever canu was requesting (you can edit ```correction/2-correction/correctReads.jobSubmit-01.sh```, manually change the memory and re-launch that job to see if it succeeds).

**Comment by zya226** [comment]
Hi Sergey,

I increased memory three times, however I still could not go through this process. I don't have WORKING cns file in result folder. They are all *.cns files. 

Here is my command in correctReads.jobSubmit-01.sh
#!/bin/sh

bsub \
  -R span[hosts=1] -M 16384 -R rusage[mem=20480] -n 10 -q long -W 24:00 -o correctReads.%I.out \
  -J "cor_GD02[1-42]" \
   \
  `pwd`/correctReads.sh 0 \
> ./correctReads.jobSubmit-01.out 2>&1


This is the command line I used before: /farline/uma_lijun_ma/software/canu/Linux-amd64/bin/canu -correct -p GD02 -d GD02_assembly merylThreads=30 maxMemory=200G ovsMemory=100G-200G ovsThreads=16 ovbMemory=200G gridOptions="-R rusage[mem=20480] -n 4 -q long -W 24:00" ovbConcurrency=16 oeaThreads=16 genomeSize=52m minReadLength=2000 minOverlapLength=500 corOutCoverage=300 corMinCoverage=2 -pacbio-raw GD02.filtered_subreads.fq.gz

**Comment by zya226** [comment]
I checked *.loadCorrectedReads.err file in correction, it is said "sqStoreBlobWriter()-- blob file '../GD02.seqStore/blobs.0003' already exists."  Is this problem?

**Comment by skoren** [comment]
That doesn't make much sense, it shouldn't be loading corrected reads if the corrected reads didn't finish. 

Do you have 42 cns files under 2-correction/results (without WORKING in the name)? If you don't then that step needs to be completed first before looking at the loading. I don't see the increased memory in correctedReads.jobSubmit-01.sh, it should have defaulted to 16gb based on your genome size. This is what I was suggesting doubling so you should make it -M 32000 or -M 64000, remove the -R rusage[mem=20480] from both the script and your canu command line, you're overwriting canu's memory/threads requests with your own which will cause issues (you also shouldn't set ovb and ovs memory so high, 32 at most is reasonable).

If you do have all 42 consensus files, we need to see what happened with the loading. What is the file contents (ls -la) of GD02.seqStore in the top-level folder (GD02_assembly)?

**Comment by zya226** [comment]
Okay, I thought increasing memory means increasing number of cores. I will change -M and let you know the result.

**Comment by zya226** [comment]
Hi Sergey,

I am done with correction part after changing the parameters as you suggested. Now I am in trimming step. However, I met new errors as below. 
-- Creating overlap store trimming/GD02.ovlStore using:
--      2 buckets
--      2 slices
--        using at most 15 GB memory each
--
-- Overlap store sorting jobs failed, tried 2 times, giving up.
--   job trimming/GD02.ovlStore.BUILDING/0001 FAILED.
I have increased memory in overlap.jobSubmit-01.sh twice, but did not work. Here is the command:
#!/bin/sh

bsub \
  -R span[hosts=1] -n 4 -M 1280000 -n 18 -q long -W 24:00 -o overlap.%I.out \
  -J "obtovl_GD02[1-108]" \
   \
  `pwd`/overlap.sh 0 \
> ./overlap.jobSubmit-01.out 2>&1

I don't know what to do. Continue increasing memory?  i have GD02.ovlStore.BUILDING under trimming folder. I guess this step has some problem.

**Comment by skoren** [comment]
Post the full log from the failing jobs, ```trimming/GD02.ovlStore.BUILDING/logs/2-sort.*.out```.

**Comment by zya226** [comment]
Sergey, 

This is still memory problem. Below is error. 

Your job looked like:

------------------------------------------------------------
# LSBATCH: User input
/farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING/scripts/2-sort.sh 0
------------------------------------------------------------

TERM_MEMLIMIT: job killed after reaching LSF memory usage limit.
Exited with exit code 137.

Another question: how could I let the default memory setting not shown in *.jobSubmit-*.sh? I have included gridOptions in my original command, but *.jobSubmit-*.sh always has default -M setting in front of parameters in gridOptions.

**Comment by zya226** [comment]
Sergey,

I increased memory to -M 2560000, however did not work. This is a lot of memory. I checked trimming/*.ovlStore.BUILDING/logs/2-sort.*.out; The file logs/2-sort.2.out showed Successfully completed. But logs/2-sort.1.out does not. 

Here is the fully report:
Found canu:
   /farline/uma_lijun_ma/software/canu/Linux-amd64/bin/canu
   Canu snapshot v1.8 +200 changes (r9410 1ccd6fe97de8cfeb265d032cd7496459b879c2b0)

Running job 1 based on LSB_JOBINDEX=1 and offset=0.

Attempting to increase maximum allowed processes and open files.
  Max processes per user limited to 2067637, no increase possible.
  Max open files limited to 262144, no increase possible.


Finding overlaps.
  found  305932106 overlaps in './R967.ovlStore.BUILDING/bucket0001/sliceSizes'.
  found  306092003 overlaps in './R967.ovlStore.BUILDING/bucket0002/sliceSizes'.

Loading  612024109 overlaps using 13.68 GB of allowed (-M) 15 GB memory.
/farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING/scripts/2-sort.sh: line 113: 65492 Killed                  $bin/ovStoreSorter -O ./R967.ovlStore.BUILDING -S ../R967.seqStore -C ./R967.ovlStore.config -f -s $jobid -M 15

------------------------------------------------------------
Sender: LSF System <lsfadmin@c07b07>
Subject: Job 4164232[1]: <ovS_R967[1-2]> in cluster <umghpcc> Exited

Job <ovS_R967[1-2]> was submitted from host <c03b03> by user <yz92a> in cluster <umghpcc> at Tue Apr 30 14:48:51 2019.
Job was executed on host(s) <10*c07b07>, in queue <long>, as user <yz92a> in cluster <umghpcc> at Tue Apr 30 14:50:39 2019.
</home/yz92a> was used as the home directory.
</farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING> was used as the working directory.
Started at Tue Apr 30 14:50:39 2019.
Terminated at Tue Apr 30 14:50:51 2019.
Results reported at Tue Apr 30 14:50:51 2019.

Your job looked like:

------------------------------------------------------------
# LSBATCH: User input
/farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING/scripts/2-sort.sh 0
------------------------------------------------------------

TERM_MEMLIMIT: job killed after reaching LSF memory usage limit.
Exited with exit code 137.

**Comment by skoren** [comment]
It only needs 15gb of memory so if it is exceeding 256, this most likely means you have disk corruption on your system. I didn't see the memory usage/description in the logs, what does LSF say the memory used was when it was killed?

Can you run ls on both bucket files (```ls -la /R967.ovlStore.BUILDING/bucket*/```)? Try removing the first bucket and manually re-running its creation (e.g. ```sh scripts/1-bucketize.sh 1```) then re-try the first sort job.

**Comment by zya226** [comment]
Sergey,

I re-run sh scripts/1-bucketize.sh 1. This is fine. However, when I ran the sort job, it was failed. Here is the error:

jdk 1.8.0_77 is located under /share/pkg/jdk/1.8.0_77
gnuplot 5.2.0 is located under /share/pkg/gnuplot/5.2.0
gcc 5.4.0 is located under /share/pkg/gcc/5.4.0

Finding overlaps.
  found  305932106 overlaps in './R967.ovlStore.BUILDING/bucket0001/sliceSizes'.
  found  306092003 overlaps in './R967.ovlStore.BUILDING/bucket0002/sliceSizes'.

Loading  612024109 overlaps using 13.68 GB of allowed (-M) 15 GB memory.
scripts/2-sort.sh: line 113: 72512 Killed                  $bin/ovStoreSorter -O ./R967.ovlStore.BUILDING -S ../R967.seqStore -C ./R967.ovlStore.config -f -s $jobid -M 15

I don't understand how this is happen. I request -M 128000 & -n 30. 

This is the job log:

Found perl:
   /share/pkg/perl/5.18.1/bin/perl
   This is perl 5, version 18, subversion 1 (v5.18.1) built for x86_64-linux-thread-multi

Found java:
   /share/pkg/jdk/1.8.0_77/bin/java
   java version "1.8.0_77"

Found canu:
   /farline/uma_lijun_ma/software/canu/Linux-amd64/bin/canu
   Canu snapshot v1.8 +200 changes (r9410 1ccd6fe97de8cfeb265d032cd7496459b879c2b0)

Running job 1 based on LSB_JOBINDEX=0 and offset=0.

Attempting to increase maximum allowed processes and open files.
  Max processes per user limited to 769216, no increase possible.
  Max open files limited to 262144, no increase possible.


------------------------------------------------------------
Sender: LSF System <lsfadmin@c36b10>
Subject: Job 4196204: <#!/bin/bash; #BSUB -n 30;#BSUB -M 128000;#BSUB -q long;#BSUB -W 12:00;#BSUB -o "/home/yz92a/output/%J.out";#BSUB -e "/home/yz9
2a/output/%J.err";module load java/1.8.0_77;#module load java/1.7.0_25;#module load canu/1.5;module load gnuplot/5.2.0;module load gcc/5.4.0;module l
oad perl/5.18.1;   #sh scripts/1-bucketize.sh 1; sh scripts/2-sort.sh 1> in cluster <umghpcc> Exited

Job <#!/bin/bash; #BSUB -n 30;#BSUB -M 128000;#BSUB -q long;#BSUB -W 12:00;#BSUB -o "/home/yz92a/output/%J.out";#BSUB -e "/home/yz92a/output/%J.err";
module load java/1.8.0_77;#module load java/1.7.0_25;#module load canu/1.5;module load gnuplot/5.2.0;module load gcc/5.4.0;module load perl/5.18.1;  
 #sh scripts/1-bucketize.sh 1; sh scripts/2-sort.sh 1> was submitted from host <ghpcc06> by user <yz92a> in cluster <umghpcc> at Wed May  1 09:59:39 
2019.
Job was executed on host(s) <8*c36b10>, in queue <long>, as user <yz92a> in cluster <umghpcc> at Wed May  1 09:59:46 2019.
                            <7*c36b09>
                            <6*c36b11>
                            <1*c36b16>
                            <4*c36b02>
                            <1*c36b12>
                            <3*c37b06>
</home/yz92a> was used as the home directory.
</farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING> was used as the working directory.
Started at Wed May  1 09:59:46 2019.
Terminated at Wed May  1 09:59:54 2019.
Results reported at Wed May  1 09:59:54 2019.

Your job looked like:

------------------------------------------------------------
# LSBATCH: User input
#!/bin/bash

#BSUB -n 30
#BSUB -M 128000
#BSUB -q long
#BSUB -W 12:00
#BSUB -o "/home/yz92a/output/%J.out"
#BSUB -e "/home/yz92a/output/%J.err"
module load java/1.8.0_77
#module load java/1.7.0_25
#module load canu/1.5
module load gnuplot/5.2.0
module load gcc/5.4.0
module load perl/5.18.1



#sh scripts/1-bucketize.sh 1

sh scripts/2-sort.sh 1

------------------------------------------------------------

TERM_MEMLIMIT: job killed after reaching LSF memory usage limit.
Exited with exit code 137.

Resource usage summary:

    CPU time :                                   3.82 sec.
   Max Memory :                                 8192 MB
    Average Memory :                             2374.00 MB
    Total Requested Memory :                     30720.00 MB
    Delta Memory :                               22528.00 MB
    Max Swap :                                   -
    Max Processes :                              5
    Max Threads :                                6
    Run time :                                   7 sec.
    Turnaround time :                            15 sec.

The output (if any) is above this job summary.



PS:

Read file </home/yz92a/output/4196204.err> for stderr output of this job.

**Comment by skoren** [comment]
You didn't post the contents of the bucket folders. That can help figure out which files are corrupt. 

Before you're re-ran the first bucketize job, did you remove the full bucket0001 folder? Try removing both bucket0001 and bucket0002, re-running them, and then running the failing sort step.

**Comment by zya226** [comment]
Hi, 

I re-ran the bucketize job and sort step as you suggested. However, it did not work. 

Below are the contents of the bucket folders:
bucket0001
-rw-rw-r-- 1 yz92a uma_lijun_ma 6044467011 May  4 05:32 slice0001
-rw-rw-r-- 1 yz92a uma_lijun_ma  149998839 May  4 05:32 slice0002
-rw-rw-r-- 1 yz92a uma_lijun_ma         24 May  4 05:32 sliceSizes

bucket0002
-rw-rw-r-- 1 yz92a uma_lijun_ma 6047461522 May  4 05:32 slice0001
-rw-rw-r-- 1 yz92a uma_lijun_ma  157912151 May  4 05:32 slice0002
-rw-rw-r-- 1 yz92a uma_lijun_ma         24 May  4 05:32 sliceSizes

In the logs folder, I have four files: 1-bucketize.1.out  1-bucketize.2.out  2-sort.1.out  2-sort.2.out. Only 2-sort.1.out does not have Successfully completed. 

Here is contents in 2-sort.1.out:
Found canu:
   /farline/uma_lijun_ma/software/canu/Linux-amd64/bin/canu
   Canu snapshot v1.8 +200 changes (r9410 1ccd6fe97de8cfeb265d032cd7496459b879c2b0)

Running job 1 based on LSB_JOBINDEX=1 and offset=0.

Attempting to increase maximum allowed processes and open files.
  Max processes per user limited to 2067637, no increase possible.
  Max open files limited to 262144, no increase possible.


Finding overlaps.
  found  305932106 overlaps in './R967.ovlStore.BUILDING/bucket0001/sliceSizes'.
  found  306092003 overlaps in './R967.ovlStore.BUILDING/bucket0002/sliceSizes'.

Loading  612024109 overlaps using 13.68 GB of allowed (-M) 15 GB memory.
/farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING/scripts/2-sort.sh: line 113: 32312 Killed                  $bin/ovStoreSorter -O ./R967.ovlStore.BUILDING -S ../R967.seqStore -C ./R96
7.ovlStore.config -f -s $jobid -M 15

------------------------------------------------------------
Sender: LSF System <lsfadmin@c02b08>
Subject: Job 4562660[1]: <ovS_R967[1-1]> in cluster <umghpcc> Exited

Job <ovS_R967[1-1]> was submitted from host <c02b08> by user <yz92a> in cluster <umghpcc> at Sat May  4 05:33:26 2019.
Job was executed on host(s) <10*c02b08>, in queue <long>, as user <yz92a> in cluster <umghpcc> at Sat May  4 05:33:29 2019.
</home/yz92a> was used as the home directory.
</farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING> was used as the working directory.
Started at Sat May  4 05:33:29 2019.
Terminated at Sat May  4 05:33:41 2019.
Results reported at Sat May  4 05:33:41 2019.

Your job looked like:

------------------------------------------------------------
# LSBATCH: User input
/farline/uma_lijun_ma/Yong/Foc_pacbio_sequencing_data/967/967_assembly/trimming/R967.ovlStore.BUILDING/scripts/2-sort.sh 0
------------------------------------------------------------

TERM_MEMLIMIT: job killed after reaching LSF memory usage limit.
Exited with exit code 137.

Resource usage summary:

    CPU time :                                   11.49 sec.
    Max Memory :                                 10240 MB
    Average Memory :                             699.00 MB
    Total Requested Memory :                     10240.00 MB
    Delta Memory :                               0.00 MB
    Max Swap :                                   -
    Max Processes :                              4
    Max Threads :                                5
    Run time :                                   11 sec.
    Turnaround time :                            15 sec.

**Comment by skoren** [comment]
The most recent run indicates you only had 10gb allocated to the sort job, it needs at least 15 (as you can see  from the log). Try reserving 20gb and trying again and post the output.

**Comment by zya226** [comment]
Hi,
After I remove the bucket folder, I could not re-ran it. It is said "Failed to open './R967.ovlStore.config' for reading: No such file or directory". And I copy this file into scripts folder, it still did not work.

**Comment by skoren** [comment]
Why did we return to the bucket step? I thought from your previous comments you had already re-run that that and were running sort but just on a lower memory node?

To make it easier, remove ```R967.ovlStore.BUILDING```, and ```R967.ovlStore*``` and just re-launch Canu. Hopefully this time your file system will not  have an issue.

**Comment by zya226** [comment]
I did this way you suggested "remove R967.ovlStore.BUILDING, and R967.ovlStore* and just re-launch Canu. ". The last errors I post are from this way, which displayed 10gb memory has been used.

**Comment by skoren** [comment]
OK, I'm lost as to what the current state of your assembly folder is. Please only try one step at a time and report back. The current list seems to include multiple mixed attempts (removing buckets and not being able to re-create them, re-creating them and sort failing through a canu run, manual sort jobs run).

Assuming you have buckets from Canu's output and the sort job failed due to the memory limit, have you already tried to request a 40gb instance and running the sort job manually? What is the contents of ```scripts/2-sort.jobSubmit*sh```?

**Comment by skoren** [comment]
Idle, if you have another error in this assembly open a new issue.
